src/index_ripper/app.py

```
@@ -0,0 +1,1516 @@
+                continue
+
+            dir_path = info.get("path", "")
+            path_segments = [seg for seg in dir_path.strip("/").split("/") if seg]
+            try:
+                target_dir, file_path = build_download_path(self.download_path, path_segments, file_name)
+            except ValueError:
+                self.log_message(f"[Download] Skipped unsafe path: {full_path}")
+                continue
+            os.makedirs(target_dir, exist_ok=True)
+
+            cancel_event = self.downloads_panel.ensure(file_path, safe_name)
```

Ruff (0.15.6)

[error] Undefined name `safe_name`

(F821)